### PR TITLE
Add javadocs jars

### PR DESCRIPTION
**Pull request by rjernst**
This change adds javadoc jars to core, test-framework and plugins. There
were a couple issues which javadoc found, but doclint did not already
find.

### CODE REVIEW

**Comment by nik9000** [comment]
Lgtm. The command line changes you made just make it ignore the missing docs, right?  I also see quiet. If there are docs warnings are they just eaten?

**Comment by rmuir** [comment]
Why is the javac compilation not failing right now? Something in the build is broken there.

**Comment by rmuir** [comment]
I suspect its gradle's non-forking compiler. Can we just fork an actual call to 'javac'? Then we know it works.

**Comment by rjernst** [comment]
I created another PR to use forking for javac: #14779. However, it does not cause failures from doclint. Since we have doclint enabled in maven, and maven did not catch the issues I found here, it must be issues with doclint itself?

**Comment by rjernst** [comment]
@nik9000 I added a comment explaining the -quiet. It was tricky to find (through lots of web searching) and is a workaround for what seems like a bug in gradle (the addStringOption method with a single parameter doesn't seem to work).

File: build.gradle [hunk 1]
```diff
@@ -96,6 +96,14 @@ subprojects {
         include 'NOTICE.txt'   
       }
     }
+    // ignore missing javadocs
+    tasks.withType(Javadoc) { Javadoc javadoc ->
+      // the -quiet here is because of a bug in gradle, in that adding a string option
+      // by itself is not added to the options. By adding quiet, both this option and
+      // the "value" -quiet is added, separated by a space. This is ok since the javadoc
+      // command already adds -quiet, so we are just duplicating it
```
Comment (nik9000) [inline build.gradle]: Yikes! So it doesn't support adding a flag, only the `-something somevalue` style? And `''` doesn't work for the second parameter? Seems worth of an upstream bug url in the comments.

**Comment by nik9000** [comment]
Fine by me. I left a comment about maybe filing an upstream bug and adding it to the comments but LGTM either way.

**Comment by rjernst** [comment]
I added a comment referencing a gradle discuss thread on the issue.

**Pull request merged**